src/workerd/api/http.c++

```diff
@@ -1249,6 +1252,14 @@ kj::Promise<DeferredProxy<void>> Response::send(
         }
       }
     }
+
+    if (!canUseWebSocketCompression()) {
+      if (outHeaders.get(kj::HttpHeaderId::SEC_WEBSOCKET_EXTENSIONS) != nullptr) {
+        // If we haven't enabled the websocket compression feature flag, strip the header from
+        // the Response.
+        outHeaders.set(kj::HttpHeaderId::SEC_WEBSOCKET_EXTENSIONS, ""_kj);
```

Comment:
Use `unset()` to remove a header. (You don't have to check if it's present first.)

Setting to an empty string means the header will be sent, but with no content.